docs/api/rtos/ThisThread.md

```diff
@@ -0,0 +1,13 @@
+## ThisThread
+
+Use the ThisThread class to control the current thread.
```

Comment:
I think this class is used only when you are controlling the current thread. If you controlling another thread, you have to use other mechanisms. I'll leave that to @kegilbert to see if we need to explicitly capture that or not.

Reply:
When would you want to control the current thread? Are there times you wouldn't?

Reply:
Thread allows you to create, join, and kill Threads. 
ThisThread allows you to control the currently running thread (Sleep, wait for flags, etc).

Reply:
`ThisThread` was designed to avoid a specific confusion with static `Thread` methods that quietly acted on the current thread, rather than any thread named. It makes that explicit. All such confusing `Thread` methods have been deprecated.

Confusion was caused because the static methods in `Thread` meant this wasn't clear:

      threadA->signal_set(1);     // signals 1 to `thread`
      Thread::signal_wait(1);     // makes this thread wait for signal 1
      threadA->signal_wait(1);    // XXX makes *this* thread wait for signal 1 - `threadA` value is ignored!

      Thread::gettid();           // reads current thread's ID
      threadA->gettid();          // XXX reads *current* thread's ID, not ThreadA's!

Those confusing static methods are now deprecated, and code would now look like

    threadA->flags_set(1);            // ("signal" changed to "flags" to match CMSIS-RTOS 2)
    ThisThread::flags_wait_any(1); 
    //threadA->flags_wait_any(1);     // this would correctly fail to compile

    threadA->get_id();
    ThisThread::get_id();             // both legal, both do what you expect.

Not quite sure what to do with that information in the documentation. That's the justification for deprecating the old API. (Together with the fact that C11 and C++11 both follow the same new model).

But for new users, just use the methods that are available depending on what you want to do - they're in the correct classes, and will do the right thing if they exist.

There are a bunch of operations which only make sense on this thread like `flags_wait_any`, so only exist in `ThisThread`; many operations would normally only be used on a different thread, so exist only in `Thread` (eg `flags_set`); a couple of operations make sense for either, so exist in both (eg `get_id`).

I guess one thing to note is that a thread may actually have no corresponding Mbed `Thread` object, because you can create a thread directly with CMSIS-RTOS APIs, or you might be `main`'s thread. You couldn't manipulate those with `Thread` methods, but `ThisThread` functions still work from inside them.